pubspec.yaml

```diff
@@ -19,7 +19,7 @@ dependencies:
 
 dev_dependencies:
   mocktail: ^0.1.3
-  test: ^1.19.5
+  test:
```

Comment:
Without looking into guidance (gotta run for the day) on this, I _think_ we'd want a constrained version?

Reply:
I tried that, but I think that will just make it more finicky with users running different versions of flutter, since the flutter sdk also depends on test. Because `pub publish --dry-run` wants us to include a specific `test` version, we have to add this — I think leaving it unconstrained here will make for a more forgiving dev environment, but I am not sure what the tradeoffs would be.

Reply:
I don't think I've heard a super good reason to keep an unconstrained version of a package in your pubspec before, but this line of thinking makes total sense to me, so I say A-okay 😄

Reply:
After perusing the package dependencies documentation and more specifically the [versioning syntax](https://dart.dev/tools/pub/dependencies#caret-syntax) it feels like it may be good practice to at least set the lower constraint and allow for upgrading across non-breaking versions. Leaving it blank allows upgrade beyond a breaking change (according to sem-ver)

Also, as it currently is, it is equivalent to specifying `any`, which is not recommended:

> The string `any` allows any version. This is equivalent to an empty version constraint, but is more explicit. Although any is allowed, we don’t recommend it.

If we change this to `^1.17.12` it will allow a user to run both flutter `2.8.1` (current lowest-supported version) all the way up to `2.10.2` without issue.

Reply:
Okay, awesome — makes sense to me!